pkg/crc/cache/cache.go

```diff
@@ -106,3 +108,15 @@ func (c *Cache) getBinary(destDir string) (string, error) {
 
 	return destPath, err
 }
+
+func (c *Cache) CheckVersion() error {
+	// Check if version string is non-empty
+	if c.version == "" {
+		return nil
+	}
+	currentVersion, err := c.getVerison()
+	if currentVersion != c.version {
+		return fmt.Errorf("%s does not have right version \n Required: %s \n Got: %s use 'crc setup' command.\n%v\n", c.binaryName, c.version, currentVersion, err)
```

Comment:
@cfergeau as of now whole Cache implementation is used as part of preflight and setup so I think this is good way to return consistent error in case of mismatch or you just want to have error till it says `required: foo but Got: bar` ?

Reply:
I'd consider adding an error type
```
type VersionMismatchError {
    ExpectedVersion string
    CurrentVersion string
}
func (e *VersionMismatchError) Error() string { return fmt.Sprintf("expected: %s but got: %s", e.ExpectedVersion, e.CurrentVersion) }
```
This way, the higher level code can easily generate `fmt.Errorf("%s does not have right version \n Required: %s \n Got: %s use 'crc setup' command.\n%v\n", c.binaryName, c.version, currentVersion, err)` if it wants.

Reply:
@cfergeau in that case in all the high level functions (caller of `CheckVersion` ) need to add same kind of string or create another variable which take this error message and append the other string.

Reply:
Not sure I understand your objection? If the higher level code is fine with a generic message, `e.Error()` will return such a message (`expected: %s but got: %s"`). If they want a more specific message more appropriate in the context of the higher level code, they can build one (ie they ignore `e.Error()` and use the `ExpectedVersion`/`CurrentVersion` fields).